e2e/test/iothub/service/RegistryManagerE2ETests.cs

```diff
@@ -144,5 +146,149 @@ public async Task RegistryManager_Query_Works()
                 await registryManager.RemoveDeviceAsync(deviceId).ConfigureAwait(false);
             }
         }
+
+        [LoggedTestMethod]
+        public async Task ModulesClient_GetModulesOnDevice()
+        {
+            int moduleCount = 5;
+            string testDeviceId = $"IdentityLifecycleDevice{Guid.NewGuid()}";
+            string[] testModuleIds = new string[moduleCount];
+            for (int i = 0; i < moduleCount; i++)
+            {
+                testModuleIds[i] = $"IdentityLifecycleModule{i}-{Guid.NewGuid()}";
+            }
+
+            Device device = null;
+            RegistryManager client = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+
+            try
+            {
+                // Create a device to house the modules
+                device = await client.AddDeviceAsync(new Device(testDeviceId)).ConfigureAwait(false);
+
+                // Create the modules on the device
+                for (int i = 0; i < moduleCount; i++)
+                {
+                    Module createResponse = await client.AddModuleAsync(
```

Comment:
why not call this `createdModule`, or something that describes what this response is?